doc/source/dev/release.rst

```diff
@@ -0,0 +1,259 @@
+    cd source && python generate_doc.py
+    cd ..
+    make html 2>&1 |\
+      grep -E "SEVERE|ERROR|WARNING" |\
+      grep -E -v "more than one target found for|__eq__|document isn't included in any toctree"
+
+  The last command builds the documentation and filters from the output all irrelevant warnings, letting through only the "proper" warnings and errors.
+  If possible, *fix these remaining issues*.
+- Glance the built documentation (usually in ``doc/_build``) for obvious errors.
+
+
+.. _dev_rel_release_branch:
+3. Make a release branch off ``master``
+---------------------------------------
+When all tests succeed and the docs are fine, start a release branch.
+**Do not touch any actual code on this branch other than indicated below!**
+
+- Create a branch off current ``master`` with the name ``release-X.Y.Z``, inserting the correct version number, of course.
+
+  .. code-block:: bash
+
+    git checkout master
+    git pull
+    git checkout -b release-X.Y.Z
+    git push -u my_fork release-X.Y.Z
+
+  Like any regular branch that should result in a PR, the release branch is pushed to a fork.
+
+
+.. _dev_rel_bump_master:
+4. Bump the ``master`` branch to the next development version
+-------------------------------------------------------------
+To ensure a larger version number for installations from the git master branch, the version number must be increased before merging the release branch.
+
+- On the ``master`` branch, change the version string in ``odl/__init__.py`` to the next revision larger than the upcoming release version, plus ``'dev0'``.
+  For example, if the release version string is ``'0.5.3'``, use ``'0.5.4.dev0'``.
+
+  To make sure you don't miss any other location (or the information here is outdated), perform a search:
+
+  .. code-block:: bash
+
+    cd doc && make clean && cd ..  # remove the local HTML doc first
+    grep -Ir "0\.5\.4" . | grep -E -v "\.git|release_notes\.rst|odl\.egg-info"
+
+- In the file ``conda/meta.yaml``, change the version string after ``version: `` to the same as above, but without the ``dev0`` tag.
+  In the example above, this would mean to change it from ``"0.5.3"`` to ``"0.5.4"``.
+
+  If necessary, change ``git_rev`` value to ``master``, although that should already be the case.
+
+- Commit the changes, using a message like ``REL: bump version to X.Y.Z.dev0``.
+- Make a PR with just this change and merge it after review.
+  It must be merged before the release branch.
+
+
+.. _dev_rel_publish:
+5. Compile and publish the release
+----------------------------------
+Back on the release branch with a ``git checkout release-X.Y.Z``, it is now time to prepare the release documents, increment the version number and make a release on GitHub.
+
+- Compile the release notes.
+  They should contain all *user-visible* changes (internal stuff like test modifications is not required) and should be summarized in one or two sentences on top, perhaps mentioning the most notable changes.
```

Comment:
Mention partially visible stuff like performance improvements, i.e. that they should be documented.